q2_diversity/tests/test_core_metrics.py

```diff
@@ -144,6 +144,84 @@ def test_core_metrics(self):
         pdt.assert_series_equal(results[1].view(pd.Series), obs_feat_exp)
         pdt.assert_series_equal(results[2].view(pd.Series), shannon_exp)
 
+    def test_core_metrics_ignore_missing_samples_false(self):
+        table = biom.Table(np.array([[150, 100, 100], [50, 100, 100]]),
+                           ['O1', 'O2'],
+                           ['S1', 'S2', 'S3'])
+        table = Artifact.import_data('FeatureTable[Frequency]', table)
+
+        metadata = Metadata(
+            pd.DataFrame({'foo': ['1', '2']},
+                         index=pd.Index(['S1', 'S2'], name='id')))
+
+        with self.assertRaisesRegex(KeyError, 'samples not included'):
```

Comment:
This looks good, but I would also include the particular sample ID that isn't included in the metadata in your `assertRaisesRegex` call (that way we're being quite specific on the error being raised including the correct missing samples).